### ISSUE TITLE
test_linkcode.test_html fails with C locale and Python 3

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1311]
test_linkcode.test_html fails with C locale and Python 3.
This problem does not occur with UTF-8 locale and does not occur with Python 2.
I use Sphinx trunk (currently revision 9237e63f42e3).

```
======================================================================
ERROR: test_linkcode.test_html
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.3/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/tmp/sphinx/tests/util.py", line 199, in deco
    func(app, *args2, **kwargs2)
  File "/tmp/sphinx/tests/test_linkcode.py", line 21, in test_html
    stuff = fp.read()
  File "/usr/lib64/python3.3/encodings/ascii.py", line 26, in decode
    return codecs.ascii_decode(input, self.errors)[0]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xe2 in position 6180: ordinal not in range(128)

----------------------------------------------------------------------
Ran 290 tests in 127.068s

FAILED (SKIP=4, errors=1)
```

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1311
- Originally reported by: Arfrever Frehtes Taifersar Arahesis
- Originally created at: 2013-11-16T20:45:44.443

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-11-17 02:06:05+00:00_

I confirmed.

```
$ export LC_ALL=C
$ tox -e py33 -- -v --tests=test_linkcode
```

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2013-11-17 03:39:50+00:00_

Fix test_linkcode.test_html fails with C locale and Python 3. Closes #1311

→ <<cset d96de9d12880d8a10e0da301366a414121d88798>>
